services/121-service/src/transaction-job-processors/transaction-job-processors.service.ts

```diff
@@ -217,30 +215,151 @@ export class TransactionJobProcessorsService {
       programId: input.programId,
       paymentNumber: input.paymentNumber,
       userId: input.userId,
-      calculatedTranserAmountInMajorUnit:
+      calculatedTransferAmountInMajorUnit:
         intersolveVisaDoTransferOrIssueCardReturnDto.amountTransferredInMajorUnit,
       financialServiceProviderId: financialServiceProvider.id,
       registration,
       oldRegistration,
       isRetry: input.isRetry,
-      status: StatusEnum.success,
+      status: TransactionStatusEnum.success,
     });
   }
 
+  public async processSafaricomTransactionJob(
+    transactionJob: SafaricomTransactionJobDto,
+  ): Promise<void> {
+    // 1. Get additional data
+    const registration = await this.getRegistrationOrThrow(
+      transactionJob.referenceId,
+    );
+    const oldRegistration = structuredClone(registration);
+    const financialServiceProvider =
+      await this.getFinancialServiceProviderOrThrow(
+        FinancialServiceProviderName.safaricom,
+      );
+
+    // 2. Check if all required properties are present. If not, create a failed transaction and throw an error.
+    for (const [name, value] of Object.entries(transactionJob)) {
+      // Define "empty" based on your needs. Here, we check for null, undefined, or an empty string.
+      if (value === null || value === undefined || value === '') {
+        const errorText = `Property ${name} is undefined`;
+        await this.createTransactionAndUpdateRegistration({
+          programId: transactionJob.programId,
+          paymentNumber: transactionJob.paymentNumber,
+          userId: transactionJob.userId,
+          calculatedTransferAmountInMajorUnit: transactionJob.transactionAmount,
+          financialServiceProviderId: financialServiceProvider.id,
+          registration,
+          oldRegistration,
+          isRetry: transactionJob.isRetry,
+          status: TransactionStatusEnum.error,
+          errorText,
+        });
+        return;
+      }
+    }
+
+    // 3. Check for existing safaricom transfer with the same originatorConversationId. This implies an unintended Redis job re-attempt.
```

Comment:
Suggest to rephrase: 

Check for existing Safaricom Transfer with the same originatorConversationId, because that means this job has already been (partly) processed. In case of a server crash, jobs that were in process are processed again.